File: ItemBuilder.java
package net.glitchtechs.starapi;

import java.util.ArrayList;
import java.util.List;
import org.bukkit.Bukkit;
import org.bukkit.Material;
import org.bukkit.attribute.Attribute;
import org.bukkit.attribute.AttributeModifier;
import org.bukkit.enchantments.Enchantment;
import org.bukkit.inventory.ItemFlag;
import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.meta.ItemMeta;

public class ItemBuilder {
    private ItemStack itemStack;

    public ItemBuilder(Material material) {
        this.itemStack = new ItemStack(material);
    }

    public ItemBuilder amount(int amount) {
        this.itemStack.setAmount(amount);
        return this;
    }

    public ItemBuilder displayName(String displayName) {
        ItemMeta itemMeta = this.getItemMeta();
        itemMeta.setDisplayName(displayName);
        this.itemStack.setItemMeta(itemMeta);
        return this;
    }

    public ItemBuilder lore(String... lore) {
        ItemMeta itemMeta = this.getItemMeta();
        List<String> loreList = new ArrayList();
        String[] var4 = lore;
        int var5 = lore.length;

        for(int var6 = 0; var6 < var5; ++var6) {
            String line = var4[var6];
            loreList.add(line);
        }

        itemMeta.setLore(loreList);
        this.itemStack.setItemMeta(itemMeta);
        return this;
    }

    public ItemBuilder enchant(Enchantment enchantment, int level) {
        ItemMeta itemMeta = this.getItemMeta();
        itemMeta.addEnchant(enchantment, level, true);
        this.itemStack.setItemMeta(itemMeta);
        return this;
    }

    public ItemBuilder attribute(Attribute attribute, AttributeModifier modifier) {
        ItemMeta itemMeta = this.getItemMeta();
        itemMeta.addAttributeModifier(attribute, modifier);
        this.itemStack.setItemMeta(itemMeta);
        return this;
    }

    public ItemBuilder flags(ItemFlag... flags) {
        ItemMeta itemMeta = this.getItemMeta();
        itemMeta.addItemFlags(flags);
        this.itemStack.setItemMeta(itemMeta);
        return this;
    }

    public ItemStack build() {
        return this.itemStack;
    }

    private ItemMeta getItemMeta() {
        ItemMeta itemMeta = this.getItemStack().getItemMeta();
        if (itemMeta == null) {
            itemMeta = Bukkit.getItemFactory().getItemMeta(this.getItemStack().getType());
            this.getItemStack().setItemMeta(itemMeta);
        }

        return itemMeta;
    }

    private ItemStack getItemStack() {
        return this.itemStack;
    }
}

File: Kotlin/src/ItemBuilder.kt
package net.glitchtechs.starapi

import org.bukkit.Bukkit
import org.bukkit.Material
import org.bukkit.attribute.Attribute
import org.bukkit.attribute.AttributeModifier
import org.bukkit.enchantments.Enchantment
import org.bukkit.inventory.ItemFlag
import org.bukkit.inventory.ItemStack
import org.bukkit.inventory.meta.ItemMeta


class ItemBuilder(material: Material?) {
    private val itemStack = ItemStack(material!!)

    fun amount(amount: Int): ItemBuilder {
        itemStack.amount = amount
        return this
    }

    fun displayName(displayName: String?): ItemBuilder {
        val itemMeta = this.itemMeta
        itemMeta!!.setDisplayName(displayName)
        itemStack.setItemMeta(itemMeta)
        return this
    }

    fun lore(vararg lore: String): ItemBuilder {
        val itemMeta = this.itemMeta
        val loreList: MutableList<String?> = ArrayList()
        val var4 = lore
        val var5 = lore.size

        for (var6 in 0 until var5) {
            val line = var4[var6]
            loreList.add(line)
        }

        itemMeta!!.lore = loreList
        itemStack.setItemMeta(itemMeta)
        return this
    }

    fun enchant(enchantment: Enchantment?, level: Int): ItemBuilder {
        val itemMeta = this.itemMeta
        itemMeta!!.addEnchant(enchantment!!, level, true)
        itemStack.setItemMeta(itemMeta)
        return this
    }

    fun attribute(attribute: Attribute?, modifier: AttributeModifier?): ItemBuilder {
        val itemMeta = this.itemMeta
        itemMeta!!.addAttributeModifier(attribute!!, modifier!!)
        itemStack.setItemMeta(itemMeta)
        return this
    }

    fun flags(vararg flags: ItemFlag?): ItemBuilder {
        val itemMeta = this.itemMeta
        itemMeta!!.addItemFlags(*flags)
        itemStack.setItemMeta(itemMeta)
        return this
    }

    fun build(): ItemStack {
        return this.itemStack
    }

    private val itemMeta: ItemMeta?
        get() {
            var itemMeta = itemStack.itemMeta
            if (itemMeta == null) {
                itemMeta = Bukkit.getItemFactory().getItemMeta(itemStack.type)
                itemStack.setItemMeta(itemMeta)
            }

            return itemMeta
        }
}